engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -179,9 +179,32 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
         } else if(allocationAlgorithm.equals("firstfitleastconsumed")){
             pools = reorderPoolsByCapacity(plan, pools);
         }
+
+        // Hardware accelerated pools are preferred for thick disks
+        if (dskCh != null && !dskCh.getProvisioningType().equals(Storage.ProvisioningType.THIN) &&
```

Comment:
i think it is better to move this disk profile checks to private method _reorderPoolsByDiskProvisioningType()_  below

Reply:
If I move that into the method I have to pass in the plan as well. I don't think it matters here. It will also impact the readability negatively as there is already alot of logic hapening inside of that method.